src/Service/Coins/CoinsCreateService.php

```diff
@@ -23,37 +25,43 @@ class CoinsCreateService
     private $em;
 
     /**
-     * @param CoinsParseService $coinsParseService
+     * @param CoinsGeckoClient $coinsGeckoClient
      * @param CoinRepository $coinRepository
      * @param EntityManagerInterface $em
      */
     public function __construct(
-        CoinsParseService $coinsParseService,
+        CoinsGeckoClient $coinsGeckoClient,
         CoinRepository $coinRepository,
         EntityManagerInterface $em
         )
     {
-        $this->coinsParseService = $coinsParseService;
+        $this->coinsGeckoClient = $coinsGeckoClient;
         $this->coinRepository = $coinRepository;
         $this->em = $em;
     }
 
-    public function create(): void
+    public function create(OutputInterface $output): void
     {
-        $parsedCoins = $this->coinsParseService->parseAll();
+        $coins = $this->coinsGeckoClient->getAll();
 
-        foreach($parsedCoins as $parsedCoin){
-            if(!$this->coinRepository->findOneBy(['coingeckoId' => $parsedCoin->id])){
+        $progressBar = new ProgressBar($output, count($coins));
+
+        $progressBar->start();
+        foreach($coins as $singleCoin){
+            if(!$this->coinRepository->findBy(['coingeckoId' => $singleCoin->id])){
```

Comment:
you were supposed to change it to find parts, not one by one, because for 12k coints this will take a long time